src/SimpleSAML/Utils/HTTP.php

```diff
@@ -1187,12 +1189,23 @@ public function submitPOSTData(string $destination, array $data): void
         if ($allowed && preg_match("#^http:#", $destination) && $this->isHTTPS()) {
             // we need to post the data to HTTP
             $this->redirect($this->getSecurePOSTRedirectURL($destination, $data));
+            return;
         }
 
         $p = new Template($config, 'post.twig');
         $p->data['destination'] = $destination;
         $p->data['post'] = $data;
+
+        // Read optional config override; default to 30s, ensure non-negative integer
+        $delay = (int) ($config->getOptionalValue('slow_post_delay_ms', 30000));
```

Comment:
Use getOptionalInteger or getOptionalIntegerRange instead and drop the cast.

Reply:
@tvdijen done.